### DISCUSSION TITLE
Error regarding Save - and CreateButton

### DISCUSSION THREAD

**Discussion by AnnaD91** [neuefische/questions#753] [Web] [answered]
Hi there, 

while working on the Modals we encountered the following error message: 
`````
Unhandled Runtime Error
Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.

Check the render method of `SaveButton`.
`````

We separated the SaveButton and the CreateButton from each other and tried to style both components as well as the buttons, links and paragraphs in our `getModalContent` function and suddenly received the error message from above even though it was working perfectly fine before. 

Please find our PR [here](https://github.com/e-rmkt/capstone-project-travel-app/pull/32). 

We would appreciate any help, thanks in advance ,

Team Tripify 🌍

**Comment by JessicaLoers** [comment, marked as answer]
**Hi @AnnaD91**
It took a while to find the two mean bugs brothers 🐛 🐛 

1. In the document `TripForm/index,` you are importing a Styled Component named `StyledModalParagraph` which doesn't exist in the file `Trip/TripForm.styled` 

🏁 **Possible solutions:**
- Add the missing Styled Component `StyledModalParagraph` to the file `TripForm.styled.js`.
- Or import the `StyledModalParagraph` component from the `Modals.styled.js` file and not the `TripForm.styled.js`

2.  Once you have fixed this bug, a new challenge will appear 😶‍🌫️
```
🚫 Warning: validateDOMNesting(...): <p> cannot appear as a descendant of <p>.
```

_This warning arises when you're trying to nest a `<p> `element inside another `<p>` element in your React components._ 

**🏁 Quick Fix**
Change the `StyledModalParagraph` to a different Element such as a `<div>`, `<span>`, or `<section>` depending on your needs.

**Reply by AnnaD91** [reply]
Wow, thank you so much @JessicaLoers for your patience to look into this tricky problem and for you great solution. Good news: it works!! 🚀🥳 Thanks again and have a nice evening :blush:
